### PR TITLE
fix: Renamed pea replica_id to pea_id

### LINKED ISSUE #1013: Change `replica_id` argument to `pea_id` argument and change PeaRoleType accordingly
**Describe the problem**
Right now, Pea has the `PeaRoleType Replica` and an argument called `replica_id`, which can be a little missleading ans in sync with the PR #699 , it shoule be adapted accordingly.

This issue is about keeping names aligned

### PR DESCRIPTION
**Pull request by HelioStrike**
Fixes #1013 

## Changes

- Edited `pea.py` to rename argument `replica_id` to `pea_id`.
- Updated `PeaRoleType`, and changed `PeaRoleType.REPLICA` to `PeaRoleType.PARALLEL` wherever it was used.

### CODE REVIEW

**Comment by JoanFM** [comment]
Hello @HelioStrike, there is no problem to open a PR before having the definitive solution, but please convert it to `draft` (there is an option in the github GUI) so that we can see that you are still working on it.

Thanks a lot for your effort!

**Comment by HelioStrike** [comment]
@JoanFM Ah, didn't know that. Done!

**Comment by bwanglzu** [comment]
hi @HelioStrike can you run this test locally to see if you break anything?
```
===End Flaky Test Report===
=========================== short test summary info ============================
FAILED tests/unit/executors/indexers/test_binary_indexer.py::test_binarypb_in_flow
====== 1 failed, 264 passed, 17 skipped, 7 warnings in 537.95s (0:08:57) =======
```

**Comment by HelioStrike** [comment]
@bwanglzu Looks like due to some recent commit, things now work for python 3.8, but a lot of things break in the 3.9 version. Looks like other PRs have the same issue.

File: jina/peapods/pea.py [hunk 1]
```diff
@@ -107,10 +107,11 @@ def __init__(self, args: Union['argparse.Namespace', Dict]):
         """ Create a new :class:`BasePea` object
 
         :param args: the arguments received from the CLI
-        :param replica_id: the id used to separate the storage of each pea, only used when ``args.separate_storage=True``
+        :param pea_id: the id used to separate the storage of each pea, only used when ``args.separate_storage=True``
         """
         super().__init__()
         self.args = args
+        self.args.pea_id = args.replica_id
```
Comment (JoanFM) [inline jina/peapods/pea.py:114]: Is it possible to get rid of replica_id as argument as well?

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by JoanFM** [comment]
recheckcla

**Comment by HelioStrike** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by HelioStrike** [comment]
@JoanFM By getting rid, do you mean changing args.replica_id to args.pea_id everywhere in the repo?

File: jina/peapods/pea.py (hunk 1 above)
Comment (HelioStrike) [reply jina/peapods/pea.py]: @JoanFM Updated! Took me way too long to figure out where `args` was coming from.

**Review by JoanFM — APPROVED** [review]

**Pull request merged**